src/sqlfluff/core/parser/segments/base.py

```diff
@@ -1245,6 +1245,34 @@ def edit(self, raw):
         """Stub."""
         raise NotImplementedError()
 
+    reflow_pre_space = False
+
+    @property
+    def pre_reflow_strs(self) -> str:
+        """TODO: Add logic for segments which should appear before this segment."""
+        str = ""
+        if self.reflow_pre_space:
+            str += " "
```

Comment:
I don't know if this approach is going to work. We kind of want to have an idea of different _grades_ of gaps:
1. Places where we _recommend_ a break - regardless of the rest of the query (e.g. we might say that a `SELECT` or `FROM` should always be at the start of the line).
2. Places where there cloud be a space _or a newline_, and there are probably explicit rankings between them. To evaluate which we use - we would take the length of the line into account (and probably also the size of the indent).
3. Place where there should _only_ ever be a space and not a newline.